src/test/utils/test.cpp

```diff
@@ -29,6 +30,48 @@ BOOST_AUTO_TEST_CASE(test_conductivity)
     }
 }
 
+BOOST_AUTO_TEST_CASE(test_ph_temperature_compensation)
+{
+    struct TestData
+    {
+        double ph_raw, temp, ph_expected;
+    };
+
+    const TestData tests[] = {
+        // Reference: https://mantech-inc.com/faq/how-does-mantech-account-for-temperature-compensation-and-correction-in-ph-measurements/
+        // PH: 4.00 at 25 C
+        {3.775, 0.0, 4.00}, {3.82, 5.0, 4.00}, {3.91, 15.0, 4.00}, {4.00, 25.0, 4.00}, 
+        {4.09, 35.0, 4.00}, {4.18, 45.0, 4.00},
+        // PH: 10.00 at 25 C
+        {10.225, 0.0, 10.00}, {10.18, 5.0, 10.00}, {10.09, 15.0, 10.00}, {10.00, 25.0, 10.00}, 
+        {9.91, 35.0, 10.00}, {9.82, 45.0, 10.00},
+        // Reference: https://www.horiba.com/usa/water-quality/support/technical-tips/bench-meters/automatic-temperature-compensation-in-ph-measurement/
+        // PH: 7.00 at 25 C
+        {7.12, 0.0, 7.00}, {7.09, 5.0, 7.00}, {7.06, 10.0, 7.00}, {7.04, 15.0, 7.00},
+        {7.02, 20.0, 7.00}, {7.00, 25.0, 7.00}, {6.98, 30.0, 7.00}, {6.98, 35.0, 7.00},
+        {6.97, 40.0, 7.00}, {6.97, 45.0, 7.00}, {6.97, 50.0, 7.00},
+        // PH: 10.01 at 25 C
+        {10.32, 0.0, 10.01}, {10.25, 5.0, 10.01}, {10.18, 10.0, 10.01}, {10.12, 15.0, 10.01},
+        {10.06, 20.0, 10.01}, {10.01, 25.0, 10.01}, {9.97, 30.0, 10.01}, {9.93, 35.0, 10.01}, 
+        {9.89, 40.0, 10.01}, {9.86, 45.0, 10.01}, {9.83, 50.0, 10.01},
+    };
+
```

Comment:
Following my previous comment:  If we're correcting a measured pH to a true pH, then we'd expect:

* For a low pH (<7.0) and a low temperature (< 25C), then the actual pH would be further from 7.0 (lower) than the measured pH, but the first test is the opposite:
  {3.775, 0.0, 4.00}

In summary, it looks to me like the `temperature_compensated_ph` function is adjusting backward as a result of the previous comment, and the tests are backward as well?  Unless I'm missing something...

Reply:
Thank you Ed!